### PR TITLE
Fix no corresponding backend roles user can update model issue

### PR DESCRIPTION
**Pull request by zane-neo**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java [hunk 1]
```diff
@@ -154,15 +154,18 @@ private void updateModelGroup(
     }
 
     private void validateRequestForAccessControl(MLUpdateModelGroupInput input, User user, MLModelGroup mlModelGroup) {
-        if (hasAccessControlChange(input) && !modelAccessControlHelper.isOwner(mlModelGroup.getOwner(), user)) {
-            throw new IllegalArgumentException("Only owner has valid privilege to perform update access control data");
+        if (hasAccessControlChange(input)) {
+            if (!modelAccessControlHelper.isOwner(mlModelGroup.getOwner(), user)) {
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java:158]: Should we also consider admin?

**Review by ylwu-amzn — DISMISSED** [review]

**Review by ylwu-amzn — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/action/model_group/TransportRegisterModelGroupAction.java [hunk 2]
```diff
@@ -153,14 +153,19 @@ public void createModelGroup(MLRegisterModelGroupInput input, ActionListener<Str
...
     private void validateRequestForAccessControl(MLRegisterModelGroupInput input, User user) {
-        ModelAccessIdentifier modelAccessIdentifier = input.getModelAccessIdentifier();
+        ModelAccessMode modelAccessMode = input.getModelAccessMode();
         Boolean isAddAllBackendRoles = input.getIsAddAllBackendRoles();
-        if (modelAccessIdentifier == null) {
-            throw new IllegalArgumentException("Invalid model access identifier");
-        } else if ((ModelAccessIdentifier.PUBLIC == modelAccessIdentifier || ModelAccessIdentifier.PRIVATE == modelAccessIdentifier)
+        if (modelAccessMode == null) {
+            if (!Boolean.TRUE.equals(isAddAllBackendRoles) && CollectionUtils.isEmpty(input.getBackendRoles())) {
+                throw new IllegalArgumentException("User must specify at least one backend role or make the model public/private");
```
Comment (rbhavna) [inline plugin/src/main/java/org/opensearch/ml/action/model_group/TransportRegisterModelGroupAction.java:160]: If we do not specify any access fields, the default should be private actually, not throw exception

Comment (zane-neo) [reply plugin/src/main/java/org/opensearch/ml/action/model_group/TransportRegisterModelGroupAction.java:160]: I think let user aware of this need more time, explicitly specify the model mode to private would be easy to understand and less confuse to user, let's make this explicit, if there's callout from user, we can change the behavior.

Comment (rbhavna) [reply plugin/src/main/java/org/opensearch/ml/action/model_group/TransportRegisterModelGroupAction.java:160]: yeah should be fine then

**Review by rbhavna — APPROVED** [review]

**Pull request merged**